File: backend/app/src/models/buildings/BuildersHut.py
from ..blueprints.Factory import Factory


class BuildersHut(Factory):
    def __init__(self, workers: list = []) -> None:
        super().__init__(name="Builders Hut", worker_slots=2, workers=workers)
        self.action_list: list[dict] =  [
                                            { "name": "Convert Worker To Builder", "action": self.convert_worker_to_builder },
                                            { "name": "Get Workers", "action": self.get_workers }
                                        ]
        
    
    def match_payload_action(self, action: str, context: list) -> dict:
        try:
            for act in self.action_list:
                if act["name"] == action:
                    return act["action"](*context)
        except Exception as e:
            return  { "success": False, "message": e }
            

    def convert_worker_to_builder(self, worker_id, settlement) -> dict:
        return self.convert_worker_to_craftsman("Builders Hut", worker_id, settlement)

File: backend_refactoring/CSharpApi/Models/BluePrints/BuildingTypes/BuildersHut.cs

using CSharpApi.Models.BluePrints.Beings;
using CSharpApi.Models.BluePrints.Locations;

namespace CSharpApi.Models.BluePrints.BuildingTypes
{
    public class BuildersHut : Factory, IBuildingList 
    {
        public string Name { get; set; } = "Builders Hut";

        public Dictionary<string, Dictionary<string, float>> Costs { get; set; } = new()
        {
            {
                "costs", new()
                {
                    { "gold", 200 },
                    { "food", 150 },
                    { "wood", 200 },
                    { "iron", 50 }
                }
            }
        };

        private List<Worker> _workers;     

        public BuildersHut(List<Worker> workers = null) : base("Builders Hut", true, 2, workers ?? [])
        {
            _workers = workers ?? [];
        }

        public object MatchPayLoadAction(string action, List<object> context)
        {
            return action switch
            {
                "Get Workers" => _workers,
                "Convert Worker To Builder" => GetWorker(),
                _ => new Dictionary<string, object>()
                    {
                        { "success", false }, { "message", "Non Executable Action" }
                    },
            };
        }

        public Dictionary<string, object> ConvertWorkerToBuilder(int workerId, object settlement)
        {
            return ConvertWorkerToCraftsman(Name, workerId, settlement);
        }
    }
}
